metricbeat/module/postgresql/replication/data.go

```diff
@@ -0,0 +1,53 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package replication
+
+import (
+	"time"
+
+	s "github.com/elastic/beats/v7/libbeat/common/schema"
+	c "github.com/elastic/beats/v7/libbeat/common/schema/mapstrstr"
+)
+
+// Based on https://www.postgresql.org/docs/9.2/static/monitoring-stats.html#PG-STAT-DATABASE-VIEW
+var schema = s.Schema{
+	"pid": c.Int("pid"),
+	"user": s.Object{
+		"id":   c.Int("usesysid", s.Optional),
+		"name": c.Str("usename"),
+	},
+	"application_name": c.Str("application_name"),
+	"client": s.Object{
+		"address":  c.Str("client_addr"),
+		"hostname": c.Str("client_hostname"),
+		"port":     c.Int("client_port", s.Optional),
+	},
+	"backend_start": c.Time(time.RFC3339Nano, "backend_start"),
+	"backend_xmin":  c.Time(time.RFC3339Nano, "backend_xmin", s.Optional),
+	"state":         c.Str("state"),
+	"sent_lsn":      c.Str("sent_lsn"),
+	"write_lsn":     c.Str("write_lsn"),
+	"flush_lsn":     c.Str("flush_lsn"),
+	"replay_lsn":    c.Str("replay_lsn"),
+	"write_lag":     c.Str("write_lag"),
```

Comment:
All the lag related fields have postgresql datatype of " Interval" , which requires 16 bytes storage. 
Here we are storing it as string. Doesn't look to be appropriate ?

Reply:
> 

How do you suggest storing?

Reply:
@shmsr Maybe you can suggest?

Reply:
Either we have to use a custom data type or maybe string only. Depends.

Example of a custom type: https://github.com/sanyokbig/pqinterval/blob/v1.1.2/interval.go#L11

https://www.postgresql.org/docs/current/datatype-datetime.html#DATATYPE-INTERVAL-INPUT

I'd suggest doing some more research before finalizing what to use. @zxcSora please let us know what are you going to use and why.

Reply:
in my opinion, sent_lsn, write_lsn, flush_lsn, replay_lsn shoud be a string, because value of this parameters it's like: 
```
sent_lsn         | 3D0/F7574390
write_lsn        | 3D0/F7574390
flush_lsn        | 3D0/F7574390
replay_lsn       | 3D0/F7574390
```
but for flush_lag, write_lag, replay_lag we need to use ```c.Time``` with HH:MM:SS.NNNNNN

Reply:
For lsn fields, string datatype looks fine.